backend/pkg/serde/json_schema.go

```diff
@@ -95,17 +102,26 @@ func (d JSONSchemaSerde) SerializeObject(ctx context.Context, obj any, _ Payload
 		return nil, fmt.Errorf("first byte indicates this it not valid JSON, expected brackets")
 	}
 
-	schema, err := d.SchemaSvc.GetJSONSchemaByID(ctx, so.schemaID)
+	schema, err := d.schemaClient.SchemaByID(ctx, int(so.schemaID))
```

Comment:
Technically wouldn't this also potentially find Proto or Avro Schemas?

We could have a helper function `JSONSchemaByID` that has the type test below? NBD but it might be useful.

Reply:
Yes it will find other schemas as well, we check the `schema.Type` right afterwards though before trying to parse the schema. The schema client is a thing caching / abstraction layer for retrieving schemas so didn't feel appropriate to put the JSOnSchemaByID there. I also like the idea of having the JSON schema specific parsing inside the respective serde